### ISSUE TITLE
Make it easier (possible?) to edit existing projects

### ISSUE THREAD

**Issue by arkid** [OpenRefine/OpenRefine#2173]
Please correct me if I am wrong but it seems impossible to edit the core connection settings of a project defined in the software.   I have setup mysql sql projects and these prompt for an SQL query before retrieving their data. I now want to slightly modify that query but I can't find any way to do so.   

IF there is an edit feature its incredibly unclear and well hidden so please make it more accessible.  Otherwise why is there not one?  Can one not be added ?

**Comment by thadguidry** [comment]
Correct, we don't allow modifying the connection for a Project after it has been imported.  The workflow is to REUSE a saved connection on a New Project with different query.

You can reuse an existing connection and then use that for your new query after clicking CONNECT:
![Capture](https://user-images.githubusercontent.com/986438/65906165-9e73f380-e387-11e9-9a59-4961637951f7.PNG)

**Comment by arkid** [comment]
Ok... just wished the initial query could be altered as well.  Although from what I see as to how OR functions under the hood I imagine that may be very hard/messy to implement?

**Comment by thadguidry** [comment]
If we give you a way to alter the initial query for that OpenRefine project, what do you want to happen with the exisiting data in your rows and columns in that project?
1. Give you a warning that all data and history will be completely removed and then updated with data from the new query?
2. No warning and just blow away your data in that project
3. Warning about overwriting with data from new query, but then do we handle this as a Transform and store your old data rows/columns in your history? or not even worry about history at all at this point? If you don't care about Undo/Redo history at all, this would be the same method as a Reuse Connection and Create New Project, in essence.  Where the only thing different is having to add new metadata or tags again.

Think about what your really wanting it to look like, feel like, work like...then re-open this issue and we can certainly talk about what it takes to work on those enhancement features.

Give it some more thought and reply with designs, sketches, notes at your convenience.
Until then we'll close this for now.

**Comment by arkid** [comment]
I can answer this quickly as an ideal solution; its number 2.   ie it warns you about what you are going to do and how it will influence the underlying data, then if you go ahead with it compares the table to before and then makes a new history entry that can be undone, complete with row/column changes saved. 

This would be very powerful and the logical result of making the change.    

If this is going to be a nightmare and too big to do (at least in the short term) I would request that it does option 1, with a big clear warning this will remove all the history of transforms etc.     

The background to my request is that I was exploring data in tables that I realised only after starting to manipulate them within OR were linked to many other tables.    I therefore wanted to join these tables and import many more columns.  this meant that the original SQL query needed to be altered to include joins. It was at this point I realised after hunting that it isn't possible to edit the query.

Sure I know you can simulate a join by using cell.cross syntax but if you have multiple joins to perform it is much slower to do than writing an sql query in my experience.  I think that is likely to be the case for a lot of people simply because they are more used to the sql syntax and can write these queries in external software to visualise the data (maybe even write the query for them) making it even faster.

**Comment by thadguidry** [comment]
@arkid ok this makes a bit more sense now as I understand the rationale on the background.  I'll reopen this as an enhancement request and keep it here in our backlog of issues.  No guarantee on when someone is willing to work on this however.  Lots of money can always help however ;-)

But you haven't detailed WHERE/HOW/BUTTONS that this hypothetical "update query" would fit into our current UI look & feel.  We would probably have to introduce a new button for that "update query" and only show it on a project that was created through the Import Database importer, otherwise not show it.  Perhaps that button could live under the Project Name but above the grid? or perhaps to the right of the project name?  Any drawings you can make for us or screenshots of how you want this to look and feel?  Attach any drawings or screenshots you wish to your reply by dragging and dropping.
